File: 392_Is_Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        
        
        
        p=0
        if(len(s)==0):
            return True
        for i in range(0,len(t)):
            if(t[i]==s[p]):
                p+=1
            if(p==len(s)):
                return True
        return False
            

File: 392_Is_Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int left = 0;
        int right = 0;
        if(s.empty())
            return true;
        while(right<t.length()){
		
            if(t[right]==s[left]){
                left++;
                right++;
			
            if(left>=s.length()){
                return true;
            }
            }
            else
                right++;
        }
        return false;
    }
};